ktor-server/ktor-server-netty/jvm/test/io/ktor/tests/server/netty/NettyEngineTest.kt

```diff
@@ -48,3 +49,21 @@ class NettySustainabilityTest :
         configuration.shareWorkGroup = true
     }
 }
+
+class NettyCustomChannelPipelineConfigurationTest :
+    HttpServerTestSuite<NettyApplicationEngine, NettyApplicationEngine.Configuration>(Netty) {
+    init {
+        enableSsl = true
+    }
+
+    override fun configure(configuration: NettyApplicationEngine.Configuration) {
+        configuration.shareWorkGroup = true
+        configuration.channelPipelineConfig = {
+            addLast("passThroughHandler", object : ChannelDuplexHandler() {
```

Comment:
Can't we check somehow that the config actually makes any difference? Like checking if the handler was invoked or something?  Should we actually run all integration tests against the modified netty server?

Reply:
yes.. good point. No need run the entire test suite. added a new test to verify the configured netty handler invocation.